### PR TITLE
Restructure travis to ease addition of more configurations

### PR DESCRIPTION
**Pull request by pilino1234**
This gives a cleaner `.travis.yml`, making it easier to add more configurations such as other compilers and linting.

### CODE REVIEW

File: ci/travis-script.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -e
+
+if [[ ${PLATFORM} == "Unix" ]]; then
+    mkdir -p build
+    cd build || exit 1
+
+    cmake -DPILIGHT_UNITTEST=1 \
+        -DCOVERALLS=ON \
+        -DCMAKE_BUILD_TYPE=Debug ..
+    make -j2
```
Comment (CurlyMoo) [inline ci/travis-script.sh]: Can you remove this dual core compilation?

File: .travis.yml [hunk 2]
```diff
@@ -4,42 +4,16 @@ sudo: true
...
-  apt:
-    packages:
-      - libpcap-dev
-      - libunwind8-dev
-      - lcov
-      - liblua5.2-0
-      - liblua5.2-dev
+    - env: PLATFORM=Unix COMPILER=gcc COVERAGE=1
+      compiler: gcc
+      os: linux
```
Comment (CurlyMoo) [inline .travis.yml]: Are there other platforms supported?

File: ci/travis-script.sh (hunk 1 above)
Comment (pilino1234) [reply ci/travis-script.sh]: Will do

File: .travis.yml (hunk 2 above)
Comment (pilino1234) [reply .travis.yml]: The other supported option is osx, which probably isn't necessary at this stage.

I was thinking about doing windows compilations too (they would need to be cross-compiled with mingw). However, I have no idea how to set up cross-compiling toolchains, so I left this out for now

Comment (CurlyMoo) [reply .travis.yml]: Windows has it's own `travis` but i need to fix windows native compilation first (which i started on rewrite).

Comment (pilino1234) [reply .travis.yml]: > Windows has it's own travis

which CI does/will it use?

Comment (CurlyMoo) [reply .travis.yml]: It's called AppVeyor https://www.appveyor.com/

**Comment by CurlyMoo** [comment]
Can you check the coveralls error in the latest travis builds:
https://travis-ci.org/pilight/pilight/builds/311483459

**Comment by pilino1234** [comment]
Should be fixed by #403

**Pull request merged**